lib/ansible/modules/cloud/amazon/iam_group.py

```diff
@@ -0,0 +1,376 @@
+            description: the date and time, in ISO 8601 date-time format, when the user was created
+            type: string
+            sample: "2017-02-08T04:36:28+00:00"
+        user_id:
+            description: the stable and unique string identifying the user
+            type: string
+            sample: AIDAIZTPY123YQRS22YU2
+        user_name:
+            description: the friendly name that identifies the user
+            type: string
+            sample: testgroup1
+        path:
+            description: the path to the user
+            type: string
+            sample: /
+'''
+
+
+import json
+
+try:
+    import boto3
+    from botocore.exceptions import ClientError, ParamValidationError
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+
+def compare_attached_group_policies(current_attached_policies, new_attached_policies):
+
+    # If new_attached_policies is None it means we want to remove all policies
+    if len(current_attached_policies) > 0 and new_attached_policies is None:
+        return False
+
+    current_attached_policies_arn_list = []
+    for policy in current_attached_policies:
+        current_attached_policies_arn_list.append(policy['PolicyArn'])
+
+    if set(current_attached_policies_arn_list) == set(new_attached_policies):
+        return True
+    else:
+        return False
+
+
+def compare_group_members(current_group_members, new_group_members):
+
+    # If new_attached_policies is None it means we want to remove all policies
+    if len(current_group_members) > 0 and new_group_members is None:
+        return False
+
+    if set(current_group_members) == set(new_group_members):
+        return True
+    else:
+        return False
+
+
+def create_or_update_group(connection, module):
+
+    params = dict()
+    params['GroupName'] = module.params.get('name')
+    managed_policies = module.params.get('managed_policy')
+    users = module.params.get('users')
+    changed = False
+
+    # Get group
+    group = get_group(connection, params['GroupName'])
+
+    # If group is None, create it
+    if group is None:
+        try:
+            group = connection.create_group(**params)
+            changed = True
+        except (ClientError, ParamValidationError) as e:
+            module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+
+    # Manage managed policies
+    current_attached_policies = get_attached_policy_list(connection, params['GroupName'])
+    if not compare_attached_group_policies(current_attached_policies, managed_policies):
+        # If managed_policies has a single empty element we want to remove all attached policies
+        if len(managed_policies) == 1 and managed_policies[0] == "":
+            for policy in current_attached_policies:
+                try:
+                    connection.detach_group_policy(GroupName=params['GroupName'], PolicyArn=policy['PolicyArn'])
+                except (ClientError, ParamValidationError) as e:
```

Comment:
need to separate ClientError and ParamValidationError as the latter doesn't have an e.response object. Both exceptions should have `exception=traceback.format_exc()` (you'll need to `import traceback` at the top)